gpbft/rebroadcast.go

```diff
@@ -0,0 +1,151 @@
+package gpbft
+
+import (
+	"fmt"
+	"time"
+)
+
+type rebroadcastState struct {
+	timeout  time.Time
+	attempts int
+
+	decideCurrentRound,
+	commitCurrentRound,
+	commitPreviousRound,
+	prepareCurrentRound,
+	preparePreviousRound,
+	convergeCurrentRound *MessageBuilder
+
+	instanceAtLatestBroadcast uint64
+	roundAtLatestBroadcast    uint64
+	phaseAtLatestBroadcast    Phase
+}
+
+func (i *instance) tryRebroadcast() error {
```

Comment:
I guess this is kind of conventional in Go, but I find spreading the methods for a single type across multiple files quite confusing. I think some of the methods could be written in against the rebroadcast state instead in which case being here would be fine.

Reply:
The alternative is giant LOC (by go standards) in one file, e.g. 1.4K LOC at `gpbft.go` and counting. 

I hear you and will move reciever funcs around. 

On a side note, I would like to reduce LOC in `gpbft.go` by breaking it into separate go files. That makes it a lot easier to navigate (for me) 😊